### DISCUSSION TITLE
How can I join where there is no foreign key?

### DISCUSSION THREAD

**Discussion by jameshfisher** [supabase/supabase#20977] [Questions] [answered]
I have these tables:

```
create table
  public.club (
    id bigint generated by default as identity,
    host_user_id uuid null,
    constraint club_host_user_id_fkey foreign key (host_user_id) references auth.users (id)
  ) tablespace pg_default;

create table
  public.profile (
    id uuid not null,
    constraint profile_pkey primary key (id),
    constraint profile_id_fkey foreign key (id) references auth.users (id)
  ) tablespace pg_default;
```

I then tried to run this query:

```
    supabase.from("club").select("id, profile(public_name)")
```

This fails with this error:

```
Searched for a foreign key relationship between 'club' and 'profile' in the schema 'public', but no matches were found.
```

The query succeeds if I change the foreign key so that `club.host_user_id` points to `profile (id)` instead of `auth.users (id)`. However, that seems like bad DB schema design?

What is the solution here? How do I join tables without a foreign key relationship? Or is that just not possible? What are the alternatives?

**Comment by GaryAustin1** [comment, marked as answer]
I always have all my data tables point to my profile.id.  Mainly so those joins work and because I want to delete the auth.user row (the user) and not necessarily have to cascade delete all the rows they were associated with.

The API can't use the auth schema at all, even for this case.

**Comment by vinumzz** [comment]
@jameshfisher What did you end up doing? I am keeping the reference to auth.users because of RLS, but can't figure out what the best way to do this is.

**Reply by GaryAustin1** [reply]
If you have a profile table you should have an id column in that referencing auth.users.  All other tables should then reference the profile id column.  You can't join thru auth.users in the REST API.

**Reply by vinumzz** [reply]
Thank you for the fast answer. I will refactor to using profiles then!
